draft-ietf-tls-tls13.md

```diff
@@ -990,26 +990,32 @@ establishment, thus providing forward secrecy.
 ## Zero-RTT Data
 
 With the Zero-RTT mode, clients can send data on their first flight
-("early data") whereby the client uses a PSK either obtained 
-out-of-band or as a ticket (i.e., one with the "early_data_info" 
-extension) from an earlier exchange to authenticate to the server. 
+("early data") whereby the client uses a PSK either obtained
+out-of-band or as a ticket (i.e., one with the "early_data_info"
+extension) from an earlier exchange to authenticate to the server.
 
-When clients use a PSK obtained out-of-band then the following 
+When clients use a PSK obtained out-of-band then the following
 information MUST be provisioned to both parties:
 
   * The PSK identity
   * The cipher suite for use with this PSK
   * The key exchange and authentication modes this PSK is allowed to be used with
-  * The Application-Layer Protocol Negotiation (ALPN) label(s)
+  * The Application-Layer Protocol Negotiation (ALPN) labels supported, if any
```

Comment:
My view is that external PSK (I think that's the term that's been used in the draft) has one set of requirements for 1-RTT with a stricter set of requirements if it is to be used for 0-RTT.

1-RTT only needs identity and hash; whereas 0-RTT needs identity, cipher suite (instead of hash), ALPN (if any), key exchange modes, and SNI (if any).  In 1-RTT, most of those parameters are negotiated; they don't need to be fixed.  (They could be, of course.)